pvlib/windspeed.py

```diff
@@ -0,0 +1,165 @@
+"""The ``windspeed`` module contains functions for calculating wind speed."""
+
+import numpy as np
+import pandas as pd
+
+
+# Values of the Hellmann exponent
+HELLMANN_SURFACE_EXPONENTS = {
+    'unstable_air_above_open_water_surface': 0.06,
+    'neutral_air_above_open_water_surface': 0.10,
+    'stable_air_above_open_water_surface': 0.27,
+    'unstable_air_above_flat_open_coast': 0.11,
+    'neutral_air_above_flat_open_coast': 0.16,
+    'stable_air_above_flat_open_coast': 0.40,
+    'unstable_air_above_human_inhabited_areas': 0.27,
+    'neutral_air_above_human_inhabited_areas': 0.34,
+    'stable_air_above_human_inhabited_areas': 0.60,
+}
+
+
+def wind_speed_at_height_hellmann(wind_speed_measured, height_measured,
+                                  height_desired, exponent=None,
+                                  surface_type=None):
+    r"""
+    Estimate wind speed for different heights.
+
+    The model is based on the power law equation by Hellmann [1]_, [2]_.
+
+    Parameters
+    ----------
+    wind_speed_measured : numeric
+        Measured wind speed. [m/s]
+
+    height_measured : float
+        The height at which the wind speed is measured. [m]
+
+    height_desired : float
+        The height at which the wind speed will be estimated. [m]
+
+    exponent : float, optional
+        Exponent based on the surface type. [-]
+
+    surface_type : string, optional
+        If supplied, overrides ``exponent``. Can be one of the following
+        (see [1]_):
+
+        * ``'unstable_air_above_open_water_surface'``
+        * ``'neutral_air_above_open_water_surface'``
+        * ``'stable_air_above_open_water_surface'``
+        * ``'unstable_air_above_flat_open_coast'``
+        * ``'neutral_air_above_flat_open_coast'``
+        * ``'stable_air_above_flat_open_coast'``
+        * ``'unstable_air_above_human_inhabited_areas'``
+        * ``'neutral_air_above_human_inhabited_areas'``
+        * ``'stable_air_above_human_inhabited_areas'``
+
+    Returns
+    -------
+    wind_speed : numeric
+        Adjusted wind speed for the desired height. [m/s]
+
+    Raises
+    ------
+    ValueError
+        If neither of ``exponent`` nor a ``surface_type`` is given.
+        If both ``exponent`` nor a ``surface_type`` is given. These parameters
+        are mutually exclusive.
+
+    KeyError
+        If the specified ``surface_type`` is invalid.
+
+    Notes
+    -----
+    The equation for calculating the wind speed at a height of :math:`h` is
+    given by the following power law equation [1]_, [2]_:
+
+    .. math::
+       :label: wind speed
+
+        windspeed_h = windspeed_{href} \cdot \left( \frac{h}{h_{ref}} \right)^a
```

Comment:
Could we use a symbol for wind speed? I think it would look neater and more readable. For example, W, U, V would be reasonable I reckon. Or you could go pick something funky from the Greek alphabet if you like
```suggestion
        W_h = W_{href} \cdot \left( \frac{h}{h_{ref}} \right)^a
```